### DISCUSSION TITLE
About lora and last update

### DISCUSSION THREAD

**Discussion by ronbere** [vladmandic/sdnext#955] [Q&A] [answered]
Since the last update, have this issue:

23:41:35-061767 WARNING  Lora failed to match keys when: D:\automatic\models\Lora\bscdb2.0.safetensors 2028
23:41:35-063768 INFO     Try using LyCORIS instead

I use this lora since many months...

Windows10, Chrome

**Comment by usigg3** [comment]
I just did a reinstall and lora still works for me.

**Comment by ronbere** [comment]
reinstall all? I do a git checkout on older commit and working too...

**Reply by usigg3** [reply]
I used --reinstall

**Comment by vladmandic** [comment]
amd gpu on windows using direct-ml? or? beacuse that's the only area i've modified today significantly.

**Reply by ronbere** [reply]
i have a nvidia gpu

**Comment by vladmandic** [comment]
can you upload that lora somewhere so i can test it? but most likely its exactly what it says.
previously if you tried to use lora which has unsupported blocks, it was ignored - so lora may-or-may-not-work, but its never going to be complete. current versions actually check each block in lora and say so. so yes, why not try it as lyco instead? most likely its not pure old-school lora to start with.

**Reply by ronbere** [reply]
Lycoris don't work at all...

23:55:54-555473 ERROR    activating extra network lyco with arguments [<modules.extra_networks.ExtraNetworkParams
                         object at 0x000001A25EEDE920>, <modules.extra_networks.ExtraNetworkParams object at
                         0x000001A2FF4997B0>]: ValueError
┌───────────────────────────────────────── Traceback (most recent call last) ─────────────────────────────────────────┐
│ D:\automatic\modules\extra_networks.py:75 in activate                                                               │
│                                                                                                                     │
│    74 │   │   try:                                                                                                  │
│ >  75 │   │   │   extra_network.activate(p, extra_network_args)                                                     │
│    76 │   │   except Exception as e:                                                                                │
│                                                                                                                     │
│ D:\automatic\extensions-builtin\a1111-sd-webui-lycoris\extra_networks_lyco.py:92 in activate                        │
│                                                                                                                     │
│   91 │   │   │   self.cache = all_lycos                                                                             │
│ > 92 │   │   lycoris.load_lycos(names, te_multipliers, unet_multipliers, dyn_dims)                                  │
│   93                                                                                                                │
│                                                                                                                     │
│ D:\automatic\extensions-builtin\a1111-sd-webui-lycoris\lycoris.py:486 in load_lycos                                 │
│                                                                                                                     │
│   485 │   │   │   if lyco is None or os.path.getmtime(lyco_on_disk.filename) > lyco.mtime:                          │
│ > 486 │   │   │   │   lyco = load_lyco(name, lyco_on_disk.filename)                                                 │
│   487                                                                                                               │
│                                                                                                                     │
│ D:\automatic\extensions-builtin\a1111-sd-webui-lycoris\lycoris.py:292 in load_lyco                                  │
│                                                                                                                     │
│   291 │   │   fullkey = convert_diffusers_name_to_compvis(key_diffusers, is_sd2)                                    │
│ > 292 │   │   key, lyco_key = fullkey.split(".", 1)                                                                 │
│   293                                                                                                               │
└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
ValueError: not enough values to unpack (expected 2, got 1)

**Comment by vladmandic** [comment]
Which extensions do you have? Post actual line from startup console.

**Reply by ronbere** [reply]
Using VENV: D:\automatic\venv
21:53:51-040928 INFO     Starting SD.Next
21:53:51-045933 INFO     Python 3.10.9 on Windows
21:53:51-141019 INFO     Version: c99c1410 Sun May 14 20:25:27 2023 -0400
21:53:51-617452 INFO     Latest published version: e737419ba43560038ae67746334a5e0e0e04c235 2023-05-15T17:05:05Z
21:53:51-619454 INFO     Setting environment tuning
21:53:51-623457 INFO     nVidia CUDA toolkit detected
21:53:55-647113 INFO     Torch 2.0.1+cu118
21:53:55-677141 INFO     Torch backend: nVidia CUDA 11.8 cuDNN 8700
21:53:55-680144 INFO     Torch detected GPU: NVIDIA GeForce RTX 3080 Ti VRAM 12287 Arch (8, 6) Cores 80
21:53:55-683146 INFO     Verifying requirements
21:53:55-696158 WARNING  Package wrong version: tqdm 4.64.1 required 4.65.0
21:53:55-699169 INFO     Installing package: tqdm==4.65.0
21:53:59-016176 INFO     Installing packages
21:53:59-359488 INFO     Running setup
21:53:59-360488 INFO     Installing repositories
21:53:59-572682 INFO     Installing submodules
21:54:02-001889 INFO     Updating submodules
21:55:18-533474 INFO     Extensions enabled: ['a1111-sd-webui-lycoris', 'clip-interrogator-ext', 'LDSR', 'Lora',
                         'multidiffusion-upscaler-for-automatic1111', 'prompt-bracket-checker', 'ScuNET',
                         'sd-dynamic-thresholding', 'sd-extension-aesthetic-scorer', 'sd-extension-steps-animation',
                         'sd-extension-system-info', 'sd-webui-controlnet', 'sd-webui-model-converter', 'seed_travel',
                         'stable-diffusion-webui-images-browser', 'stable-diffusion-webui-rembg', 'SwinIR',
                         'a1111-sd-webui-tagcomplete', 'canvas-zoom', 'model-keyword', 'openOutpaint-webUI-extension',
                         'openpose-editor', 'posex', 'sd-canvas-editor', 'sd-civitai-browser', 'SD-CN-Animation',
                         'sd-dynamic-prompts', 'sd-model-preview', 'sd-webui-3d-open-pose-editor',
                         'sd-webui-additional-networks', 'sd-webui-ar', 'sd-webui-aspect-ratio-helper',
                         'sd-webui-photopea-embed', 'sd-webui-regional-prompter', 'sd-webui-supermerger',
                         'sd_dreambooth_extension', 'stable-diffusion-webui-promptgen',
                         'stable-diffusion-webui-Prompt_Generator', 'stable-diffusion-webui-two-shot',
                         'ultimate-upscale-for-automatic1111']
21:55:18-544485 INFO     Updating Wiki
21:55:19-962774 INFO     Running extension preloading
21:55:19-964775 INFO     Server arguments: []
D:\automatic\venv\lib\site-packages\pkg_resources\__init__.py:123: PkgResourcesDeprecationWarning: llow is an invalid version and will not be supported in a future release
  warnings.warn(
No module 'xformers'. Proceeding without it.
21:55:25-614616 INFO     Libraries loaded
21:55:26-249208 INFO     Available models: F:\Telechargements\SUPER SD 2.0
                         Dependencies\stable-diffusion-webui\models\Stable-diffusion 170
ControlNet v1.1.173
ControlNet v1.1.173
21:55:30-399145 INFO     Running extension preloading
21:55:30-639363 INFO     Libraries loaded
[AddNet] Updating model hashes...
0it [00:00, ?it/s]
[AddNet] Updating model hashes...
0it [00:00, ?it/s]
21:55:32-570133 INFO     Loading UI theme: gradio/default
21:55:36-641148 INFO     Available models: F:\Telechargements\SUPER SD 2.0
                         Dependencies\stable-diffusion-webui\models\Stable-diffusion 170
CUDA SETUP: Loading binary D:\automatic\venv\lib\site-packages\bitsandbytes\libbitsandbytes_cudaall.dll...

Thanks for being a Gradio user! If you have questions or feedback, please join our Discord server and chat with us: https://discord.gg/feTf9x3ZSB
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
21:55:41-374259 INFO     Local URL: http://127.0.0.1:7860/
21:55:41-376261 INFO     API Docs: http://127.0.0.1:7860/docs
21:55:41-378263 INFO     Initializing middleware
21:55:41-860008 INFO     Setting Torch parameters: dtype=torch.float16 vae=torch.float16 unet=torch.float16

**Reply by vladmandic** [reply]
Soooo many.
Rename extensions folder and confirm if problem exists or not - you can always rename it back. 
And if that's it, add extensions one by one until you find problematic one.

Honestly, that is the absolutely first thing to check with any problem.

**Reply by ronbere** [reply]
what do u mean? rename extensions folder and add the new path?

**Reply by vladmandic** [reply, marked as answer]
Rename extensions folder so none of your extensions get loaded. Then try Lora.
You can always rename folder back later.
Testing anything with this many extensions enabled in parallel is pointless.

**Comment by ronbere** [comment]
working fine on Version: 6f976c35, but this is an old commit

**Comment by ronbere** [comment]
ok, finally i stay on old commit working fine with all extensions, loras and lycoris : https://github.com/vladmandic/automatic/commit/6f976c358f6f2db4d23af0c210f4dbb01f8a24d4

**Comment by vladmandic** [comment]
@ronbere you said you're staying on the 2 week old commit which is fine as your choice - but if there are issues, please report them so i can attempt to fix. last we spoke you had a huge list of extensions and some are known to cause problems, so i've asked to temporarily disable them to confirm issue is not in core repo.

**Reply by ronbere** [reply]
you are right, given the work provided, I will try to give the errors by activating the extensions one by one.
Sorry

**Reply by ronbere** [reply]
ok last update, rename my extensions folder give me this :
Using VENV: D:\automatic\venv
18:58:16-594556 INFO     Starting SD.Next
18:58:16-599139 INFO     Python 3.10.9 on Windows
18:58:16-888877 INFO     Version: 554f2629 Wed May 17 08:46:57 2023 -0400
18:58:18-157731 INFO     Setting environment tuning
18:58:18-161726 INFO     nVidia CUDA toolkit detected
18:58:35-241677 INFO     Torch 2.0.1+cu118
18:58:35-272543 INFO     Torch backend: nVidia CUDA 11.8 cuDNN 8700
18:58:35-274544 INFO     Torch detected GPU: NVIDIA GeForce RTX 3080 Ti VRAM 12287 Arch (8, 6) Cores 80
18:58:35-276546 INFO     Verifying requirements
18:58:35-297158 WARNING  Package wrong version: tqdm 4.64.1 required 4.65.0
18:58:35-299160 INFO     Installing package: tqdm==4.65.0
18:58:42-955263 WARNING  Package wrong version: diffusers 0.14.0 required 0.16.1
18:58:42-957264 INFO     Installing package: diffusers==0.16.1
18:58:45-604279 INFO     Installing packages
18:58:45-651322 INFO     No changes detected: Quick launch active
18:58:45-655325 INFO     Running extension preloading
18:58:45-656326 INFO     Server arguments: []
No module 'xformers'. Proceeding without it.
18:59:44-918233 INFO     Libraries loaded
18:59:49-646853 INFO     Available models: F:\Telechargements\SUPER SD 2.0
                         Dependencies\stable-diffusion-webui\models\Stable-diffusion 170
ControlNet v1.1.173
ControlNet v1.1.173
19:00:01-778464 INFO     Loading UI theme: gradio/default
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
19:00:08-365117 INFO     Local URL: http://127.0.0.1:7860/
19:00:08-367120 INFO     API Docs: http://127.0.0.1:7860/docs
19:00:08-368119 INFO     Initializing middleware
19:00:08-729564 INFO     Setting Torch parameters: dtype=torch.float16 vae=torch.float16 unet=torch.float16
Loading weights: F:\Telechargements\SUPER SD 2.0 Dependencies\stable-diffusion-webui\models\Stable-diffusion\realisti…
LatentDiffusion: Running in eps-prediction mode
DiffusionWrapper has 859.52 M params.
19:00:09-481493 INFO     Model created from config: D:\automatic\configs\v1-inference.yaml
Loading weights: F:\Telechargements\SUPER SD 2.0 Dependencies\stable-diffusion-webui\models\VAE\vae-ft-mse-840000-ema…
19:00:11-849645 INFO     Applying scaled dot product cross attention optimization

**Reply by ronbere** [reply]
and ,

Loading weights: D:\automatic\models\Lora\bscdb2.0.safetensors ------------------------------------- 0.0/4.1 GB -:--:--
19:00:54-204753 ERROR    loading Lora D:\automatic\models\Lora\bscdb2.0.safetensors: ValueError
┌───────────────────────────────────────── Traceback (most recent call last) ─────────────────────────────────────────┐
│ D:\automatic\extensions-builtin\Lora\lora.py:222 in load_loras                                                      │
│                                                                                                                     │
│   221 │   │   │   │   try:                                                                                          │
│ > 222 │   │   │   │   │   lora = load_lora(name, lora_on_disk.filename)                                             │
│   223 │   │   │   │   except Exception as e:                                                                        │
│                                                                                                                     │
│ D:\automatic\extensions-builtin\Lora\lora.py:144 in load_lora                                                       │
│                                                                                                                     │
│   143 │   for key_diffusers, weight in sd.items():                                                                  │
│ > 144 │   │   key_diffusers_without_lora_parts, lora_key = key_diffusers.split(".", 1)                              │
│   145 │   │   key = convert_diffusers_name_to_compvis(key_diffusers_without_lora_parts, is_sd2                      │
└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
ValueError: not enough values to unpack (expected 2, got 1)
100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [00:09<00:00,  2.08it/s]

**Reply by ronbere** [reply]
the size is quite large, but it works perfectly on the old commit

**Comment by ronbere** [comment]
ok, i rename extensions_builtin too

Using VENV: D:\automatic\venv
19:13:29-197780 INFO     Starting SD.Next
19:13:29-201797 INFO     Python 3.10.9 on Windows
19:13:29-277440 INFO     Version: 554f2629 Wed May 17 08:46:57 2023 -0400
19:13:29-644354 INFO     Setting environment tuning
19:13:29-648350 INFO     nVidia CUDA toolkit detected
19:13:30-872273 INFO     Torch 2.0.1+cu118
19:13:30-897583 INFO     Torch backend: nVidia CUDA 11.8 cuDNN 8700
19:13:30-899585 INFO     Torch detected GPU: NVIDIA GeForce RTX 3080 Ti VRAM 12287 Arch (8, 6) Cores 80
19:13:30-901587 INFO     Verifying requirements
19:13:30-907593 INFO     Installing packages
19:13:30-948630 INFO     No changes detected: Quick launch active
19:13:30-951633 INFO     Running extension preloading
19:13:30-952633 INFO     Server arguments: []
No module 'xformers'. Proceeding without it.
19:13:35-170640 INFO     Libraries loaded
Creating folder: D:\automatic\extensions-builtin
19:13:35-498912 INFO     Available models: F:\Telechargements\SUPER SD 2.0
                         Dependencies\stable-diffusion-webui\models\Stable-diffusion 170
19:13:35-735658 INFO     Loading UI theme: gradio/default
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
19:13:36-279357 INFO     Local URL: http://127.0.0.1:7860/
19:13:36-280358 INFO     API Docs: http://127.0.0.1:7860/docs
19:13:36-282360 INFO     Initializing middleware
19:13:36-577405 INFO     Setting Torch parameters: dtype=torch.float16 vae=torch.float16 unet=torch.float16
Loading weights: F:\Telechargements\SUPER SD 2.0 Dependencies\stable-diffusion-webui\models\Stable-diffusion\realisti…
LatentDiffusion: Running in eps-prediction mode
DiffusionWrapper has 859.52 M params.
19:13:37-321317 INFO     Model created from config: D:\automatic\configs\v1-inference.yaml
Loading weights: F:\Telechargements\SUPER SD 2.0 Dependencies\stable-diffusion-webui\models\VAE\vae-ft-mse-840000-ema…
19:13:39-344155 INFO     Applying scaled dot product cross attention optimization

**Comment by ronbere** [comment]
Skipping unknown extra network: lora
100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [00:09<00:00,  2.21it/s]
normal i think...
